### ISSUE TITLE
multi-page navigation extraction

### ISSUE THREAD

**Issue by nileio** [oxpath/oxpath-all#5]
Hi Rslan,
I am trying to build a wrapper which extracts data in the following sudo way
Page 1 : Search Page / Click
|_   Page 2 : Results Page (Results Page Iteration)
       |_  Page 3 : For Each Result entry , a Details Result Page (Click)
This is a typical navigation.

Trying to achieve the above using the following wrapper (removed some lines for brevity):
```
doc('http://www.somesite.com/')
    //input[@name='where']/{"Montreal" /}
        //button[@class='search-button']/{click /}
        /(//a[.#='Next']/{click /})*{0,1}
          //article/div[@class='listingInfo']
          :<RESULT>
                [?.//div[@class='propertyStats']//p[@class='type']:<saletype=string(.)>]
                [.//div//a[@class='name']:<addresstext=string(.)>]
                [.//div[@class='buttons']//a[@class='detailsButton']/{click /}//div[@id='DetailedContent']
                :<ITEMDETAILS>
                    [ .//div[@id='listing_header']/span:<propertyid=string(.)>]
                    [ .//div[@id='listing_header']/h1:<address=string(.)>]
                    [ .//li[@class='price']/p[1]:<price=string(.)>]
               ]
```
Notice the closing bracket at the end of the file. The above wrapper works with JDBCOutput handler and produces all expected results (with little performance issues), however, the above wrapper does not work and throws an exception with XMLOutputHandler.

With some testing, If I modify the above wrapper and place the Closing Bracket  at the end of the line of the 'detailsButton' click instead of at the end of the file, no exception is then raised with XMLOuput, BUT the `<ITEMDETAILS> `section is never then executed, and none of the attributes in the `<ITEMDETAILS> `section is extracted.
In other words, 
`[.//div[@class='buttons']//a[@class='detailsButton']/{click /}//div[@id='DetailedContent'] ]`

So on the above issue I have few questions:

- Where 'should' the opening and closing brackets exist in the example above for nested pages?
- Typically, I should have a single wrapper which can be used for any output handler. Is this achievable ? or should a wrapper be written for a specific handler ?
- If I like to extract the details page in the example above with XMLOutputHandler, how would I achieve that (given that the click action retrieves a new page)?

Thank you for your help!

**Comment by ruslanrf** [comment]
It looks like a bug in the XML output handler. Can you send me the original example, so I can test it with a newer version of the XML output handler?

**Comment by nileio** [comment]
hi ruslan,
I could not replicate the problem because I was changing lots of wrappers, and now it seems it is working !! If I find an issue I will report back. so far the tutorial was very helpful..
